src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -1043,22 +1049,23 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 	}
 
     noDuplicateCompetencies(): {duplicateCompetency: Boolean} | null {
-        if (this.duplicateCompetency)
+        if (this.checkDuplicateCompetency())
             return { duplicateCompetency: true };
     }
 
-    get duplicateCompetency(): String | null {
+    checkDuplicateCompetency(): String | null {
         if (!this.badgeClassForm) return null;
 
-        const hand = this.badgeClassForm.controls.competencies.value.
+        const storedCompetencies = this.badgeClassForm.controls.competencies.value.
             // Hand competencies get added automatically at submitting
             //filter(c => c.added).
             map(c => c.name);
-        const ai = this.badgeClassForm.controls.aiCompetencies.value.
-            filter(c => c.selected).
-            map((c,i) => this.aiCompetenciesSuggestions[i].preferred_label);
+		
+		const newSelectedCompetencies = this.badgeClassForm.controls.aiCompetencies.value
```

Comment:
Why did you rename the variables? I find the new names quite misleading, since the `storedCompetencies` are not (only) stored competencies, but also those selected by hand. Also, `newSelectedCompetencies` are not all new selected competencies, but only the AI competencies.

Reply:
I updated them as they were quite incomplete/misleading. you are right `storedCompetencies` includes those added by hand. I've updated both variables to be more meaningful.